src/taskgraph/transforms/release_notifications.py

```diff
@@ -6,11 +6,36 @@
 """
 from string import Formatter
 
+from voluptuous import ALLOW_EXTRA, Any, Optional, Required
+
 from taskgraph.transforms.base import TransformSequence
-from taskgraph.util.schema import resolve_keyed_by
-from taskgraph.util.scriptworker import get_release_config
+from taskgraph.util.schema import Schema, optionally_keyed_by, resolve_keyed_by
+
+RELEASE_NOTIFICATIONS_SCHEMA = Schema(
+    {
+        Optional("notifications"): {
+            Required("emails"): optionally_keyed_by("project", "level", [str]),
+            Required("subject"): str,
+            Required("message"): str,
+            Optional("status-types"): [
+                Any(
+                    "on-completed",
+                    "on-defined",
+                    "on-exception",
+                    "on-failed",
+                    "on-pending",
+                    "on-resolved",
+                    "on-running",
+                )
+            ],
+        },
+    },
+    extra=ALLOW_EXTRA,
```

Comment:
Oh neat, I didn't know about this. I always used the `Extra: object` trick (not sure if there's any difference between the two?)

But yeah, big +1 on only validating the keys that are actually used in the transform!